test/system/src/test/java/io/pravega/test/system/ByteClientTest.java

```diff
@@ -0,0 +1,176 @@
+            .streamName(STREAM)
+            .scalingPolicy(scalingPolicy).build();
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private ScheduledExecutorService writerExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "byte-writer");
+    private ScheduledExecutorService readerExecutor = ExecutorServiceHelpers.newScheduledThreadPool(2, "byte-reader");
+
+
+    /**
+     * This is used to setup the services required by the system test framework.
+     *
+     * @throws MarathonException When error in setup.
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+        streamManager = StreamManager.create(controllerURI);
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM, config));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+        ExecutorServiceHelpers.shutdown(writerExecutor);
+        ExecutorServiceHelpers.shutdown(readerExecutor);
+    }
+
+    /**
+     * This test verifies the correctness of basic read/write functionality of {@link ByteStreamClient}.
+     */
+    @Test
+    public void byteClientTest() throws IOException {
+        @Cleanup
+        ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
+        ControllerImpl controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(ClientConfig.builder()
+                        .controllerURI(controllerURI).build()).build(),
+                connectionFactory.getInternalExecutor());
+        @Cleanup
+        ClientFactory clientFactory = new ClientFactoryImpl(SCOPE, controller);
+        log.info("Invoking byteClientTest test with Controller URI: {}", controllerURI);
+
+        ByteStreamClient byteStreamClient = clientFactory.createByteStreamClient();
+        @Cleanup("closeAndSeal")
+        ByteStreamWriter writer = byteStreamClient.createByteStreamWriter(STREAM);
+        @Cleanup
+        ByteStreamReader reader = byteStreamClient.createByteStreamReader(STREAM);
+
+        for (int i = 1; i <= MAX_PAYLOAD_SIZE; i *= 10) {
+            final int payloadSize = i;
+            // Create the synthetic payload for the write.
+            byte[] payload = new byte[payloadSize];
+            byte[] readBuffer = new byte[payloadSize];
+            RandomFactory.create().nextBytes(payload);
```

Comment:
Create the factory outside of the loop and call next bytes inside of the loop.

Reply:
Done.